docs/user-guide/lazy-loading.md

```diff
@@ -0,0 +1,88 @@
+# Lazy Loading
+
+> [!WARNING]
+> This is an experimental option and may not work as expected with all plugins.
+> The API may change without notice. > Please report any issues you encounter.
+
+When the lazy loading functionality is enabled, the plugin's configuration is
+routed to the enabled lazy provider. Options mentioned will be in relation to
+the `${namespace}.${name}.lazyLoad` context.
+
+## Supported Lazy Loading Providers
+
+It is recommended to become familiar with lazy loading strategies supported by
+your provider in their upstream documentation and the corresponding plugin's
+support.
+
+- [Lz.n](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
+
+## Enabling Lazy Loading Per Plugin
+
+There are a couple different ways to enable lazy loading for a particular
+plugin.
+
+- `enable` set to `true`.
+- `settings` containing some configuration.
+
+If you are just wanting to store potential configuration without enabling it,
+you can explicitly disable it setting `enable = false`.
+
+When a plugin has lazy loading enabled, the Lua configuration for your plugin
+will be passed along to the corresponding lazy provider instead. This behavior
+can be overridden by using the `settings` option that will be passed to the
+provider.
+
+For `Lz.n`, that is the `settings.after` option.
+
+## Configuring Triggers
+
+Currently, you need to define the trigger conditions in which a plugin will be
+loaded. This is done through the `settings` option.
```

Comment:
I'd suggest moving away from "Currently".

Maybe one-day we could decide to offer some sane defaults, but I also wonder if that'd end up being an anti-pattern.

https://github.com/nix-community/nixvim/issues/2632#issuecomment-2540912650